spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkUtils.java

```diff
@@ -0,0 +1,210 @@
+        Schema.Field inputField = inputSchema.getField(field);
+        if (inputField == null) {
+          throw new IllegalArgumentException(String.format("Field %s does not exists in the input schema", field));
+        }
+      }
+    } else {
+      if (!Strings.isNullOrEmpty(featuresToExclude)) {
+        excludeFeatures.addAll(Lists.newArrayList(Splitter.on(',').trimResults().split(featuresToExclude)));
+      }
+      for (Schema.Field field : inputSchemaFields) {
+        String fieldName = field.getName();
+        if (!fieldName.equals(labelField) && (excludeFeatures.isEmpty() || !excludeFeatures.contains(fieldName))) {
+          fields.add(fieldName);
+        }
+      }
+    }
+    return fields;
+  }
+
+  /**
+   * Get the getCategoricalFeatureInfo using the feature to cardinality mapping provided by the user.
+   *
+   * @param inputSchema        schema of the received record.
+   * @param featuresToInclude  features to be used for training/prediction.
+   * @param featuresToExclude  features to be excluded when training/predicting.
+   * @param labelField         field containing the prediction values.
+   * @param cardinalityMapping feature to cardinality mapping specified for categorical features.
+   * @return categoricalFeatureInfo for categorical features.
+   */
+  public static Map<Integer, Integer> getCategoricalFeatureInfo(Schema inputSchema, String featuresToInclude,
+                                                                String featuresToExclude, String labelField,
+                                                                String cardinalityMapping) {
+    List<String> featureList = getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+    Map<Integer, Integer> outputFieldMappings = new HashMap<>();
+
+    if (!Strings.isNullOrEmpty(cardinalityMapping)) {
+      for (String field : Splitter.on(',').trimResults().split(cardinalityMapping)) {
+        String[] value = field.split(":");
+        if (value.length == 2) {
+          try {
+            outputFieldMappings.put(featureList.indexOf(value[0]), Integer.parseInt(value[1]));
+          } catch (NumberFormatException e) {
+            throw new IllegalArgumentException(
+              String.format("Invalid cardinality %s. Please specify valid integer for cardinality", value[1]));
+          }
+        } else {
+          throw new IllegalArgumentException(
+            String.format("Invalid categorical feature mapping '%s'. Please make sure it is in the format " +
+                            "'feature':'cardinality'.", field));
+        }
+      }
+    }
+    return outputFieldMappings;
+  }
+
+  /**
+   * Validate prediction field value for Predictor.
+   *
+   * @param inputSchema     schema of the received record.
+   * @param predictionField field on which to set the prediction.
+   */
+  public static void validatePredictionFieldForRegressor(Schema inputSchema, String predictionField) {
```

Comment:
Not a big fan of this kind of one line helper, as it actually reduces readability of the caller code. Get rid of it please.